src/worker.js

```diff
@@ -81,23 +62,54 @@ var wasmOffsetData;
 this.onmessage = function(e) {
   try {
     if (e.data.cmd === 'load') { // Preload command that is called once per worker to parse and load the Emscripten code.
+#if MINIMAL_RUNTIME
+#if MODULARIZE_INSTANCE
+      var imports = Module;
+#else
+      var imports = {};
+#endif
+#endif
+
+#if !WASM_BACKEND
+      // Initialize the thread-local field(s):
+#if MINIMAL_RUNTIME
+      var imports = {};
+#endif
+#endif
+
       // Initialize the global "process"-wide fields:
+#if !MINIMAL_RUNTIME
       Module['DYNAMIC_BASE'] = e.data.DYNAMIC_BASE;
-      Module['DYNAMICTOP_PTR'] = e.data.DYNAMICTOP_PTR;
+#endif
+
+#if USES_DYNAMIC_ALLOC
+      {{{ makeAsmImportsAccessInPthread('DYNAMICTOP_PTR') }}} = e.data.DYNAMICTOP_PTR;
```

Comment:
this is new - is it only needed for minimal_runtime?

Reply:
Hmm, not sure what you mean? Importing DYNAMICTOP_PTR is needed for both regular and minimal runtime?